android_10/src/org/ros/android/RosActivity.java

```diff
@@ -98,17 +148,44 @@ public NodeMainExecutorServiceListener getServiceListener()
 
   };
 
+  /**
+   * Standard constructor.
+   * Use this constructor to proceed using the standard {@link MasterChooser}.
+   * @param notificationTicker Title to use in Ticker notifications.
+   * @param notificationTitle Title to use in notifications.
+     */
   protected RosActivity(String notificationTicker, String notificationTitle) {
     this(notificationTicker, notificationTitle, null);
   }
 
+  /**
+   * Custom Master URI constructor.
+   * Use this constructor to skip launching {@link MasterChooser}.
+   * @param notificationTicker Title to use in Ticker notifications.
+   * @param notificationTitle Title to use in notifications.
+   * @param customMasterUri URI where to connect to.
+     */
   protected RosActivity(String notificationTicker, String notificationTitle, URI customMasterUri) {
     super();
     this.notificationTicker = notificationTicker;
     this.notificationTitle = notificationTitle;
     nodeMainExecutorServiceConnection = new NodeMainExecutorServiceConnection(customMasterUri);
   }
 
+  /**
+   * Custom MasterChooser constructor.
+   * Use this constructor to specify which {@link Activity} should be started in place of {@link MasterChooser}.
+   * @param notificationTicker Title to use in Ticker notifications.
+   * @param notificationTitle Title to use in notifications.
+   * @param activity {@link Activity} to launch instead of {@link MasterChooser}.
+   * @param requestCode Code to be sent to the specified {@link Activity}.
```

Comment:
maybe expand a tiny bit ... code to be send to the given activity at what time? for what?